packages/ui-components/src/MetadataEditor.tsx

```diff
@@ -0,0 +1,197 @@
+    this.handleLanguageChange = this.handleLanguageChange.bind(this);
+  }
+
+  renderCreateLanguageOption = (
+    query: string,
+    active: boolean,
+    handleClick: React.MouseEventHandler<HTMLElement>
+  ): React.ReactElement => (
+    <MenuItem
+      icon="add"
+      text={`Create "${query}"`}
+      active={active}
+      onClick={handleClick}
+      shouldDismissPopover={false}
+    />
+  );
+
+  filterLanguage: ItemPredicate<string> = (
+    query,
+    language,
+    _index,
+    exactMatch
+  ) => {
+    const normalizedTitle = language.toLowerCase();
+    const normalizedQuery = query.toLowerCase();
+
+    if (normalizedQuery === normalizedTitle) {
+      return normalizedTitle === normalizedQuery;
+    } else {
+      return `${language}`.indexOf(normalizedQuery) >= 0;
+    }
+  };
+
+  handleNameChange(event: any): void {
+    this.metadata.name = event.nativeEvent.srcElement.value;
+  }
+
+  handleDescriptionChange(event: any): void {
+    this.metadata.description = event.nativeEvent.srcElement.value;
+  }
+
+  handleLanguageChange = (language: string): void => {
+    this.metadata.language = language;
+    this.update();
+  };
+
+  onAfterShow(): void {
+    if (!this.editor) {
+      this.editor = this.editorFactory({
+        host: document.getElementById('code:' + this.metadata.name),
+        model: new CodeEditor.Model({ value: this.metadata.code })
+      });
+    }
+  }
+
+  onItemSelect(language: string, options: any): void {
+    this.metadata.language = language;
+    this.update();
+  }
+
+  itemRenderer(language: string, options: any): React.ReactElement {
+    return (
+      <Button
+        onClick={options.handleClick}
+        key={language}
+        text={language}
+      ></Button>
+    );
+  }
+
+  render(): React.ReactElement {
+    const nameInput = (
+      <InputGroup
+        onChange={this.handleNameChange}
+        defaultValue={this.metadata.name}
+        type="text-input"
+      />
+    );
+    const descriptionInput = (
+      <InputGroup
+        onChange={this.handleDescriptionChange}
+        defaultValue={this.metadata.description}
+        type="text-input"
+      />
+    );
+    const languageInput = (
+      <Select
+        items={defaultLanguages}
+        itemPredicate={this.filterLanguage}
+        createNewItemFromQuery={(newLanguage: string): string => {
+          return newLanguage;
+        }}
+        createNewItemRenderer={this.renderCreateLanguageOption}
+        onItemSelect={this.handleLanguageChange}
+        itemRenderer={this.itemRenderer}
+      >
+        <Button
+          icon="code"
+          rightIcon="caret-down"
+          text={
+            this.metadata.language != ''
+              ? this.metadata.language
+              : '(No selection)'
+          }
+        />
+      </Select>
+    );
+    return (
+      <div className={ELYRA_METADATA_EDITOR_CLASS}>
+        <h1> Edit {this.metadata.name} Metadata </h1>
```

Comment:
Not sure about this header content, maybe 'New code snippet' if we are creating a new one.
The text size is a bit too large as well.